src/modules/dashboard/components/Extensions/ExtensionSetup.tsx

```diff
@@ -76,11 +76,13 @@ const ExtensionSetup = ({
           const formattedPayload = {};
           initializationParams?.map(({ paramName }) => {
             if (endsWith(paramName, 'Period')) {
-              formattedPayload[paramName] = payload[paramName] * 3600; // Seconds in 1 hour
+              formattedPayload[paramName] = new Decimal(payload[paramName])
+                .mul(3600) // Seconds in 1 hour
+                .toFixed(0, Decimal.ROUND_HALF_UP);
             } else {
-              formattedPayload[paramName] = bigNumberify(
-                payload[paramName],
-              ).mul(bigNumberify(10).pow(16));
+              formattedPayload[paramName] = new Decimal(payload[paramName])
+                .mul(new Decimal(10).pow(16))
```

Comment:
Why did you use the `Decimal` instead of `bigNumberify` here? I thought that the problem was just for the periods.

Reply:
We also use `Decimal` here due to one of the reasons we use it above,  because `bigNumberify` doesn't accept decimal values as valid values, so if someone decided to input `50.5%` then that would throw an error